packages/client-config/src/unified_client_config_generator.test.ts

```diff
@@ -107,5 +110,49 @@ void describe('UnifiedClientConfigGenerator', () => {
       };
       assert.deepStrictEqual(result, expectedClientConfig);
     });
+
+    void it('throws user error if there are overlapping values', async () => {
+      const customOutputs: Partial<ClientConfig> = {
+        aws_user_pools_id: 'overrideUserPoolId',
+      };
+      const stubOutput: UnifiedBackendOutput = {
+        [authOutputKey]: {
+          version: '1',
+          payload: {
+            identityPoolId: 'testIdentityPoolId',
+            userPoolId: 'testUserPoolId',
+            webClientId: 'testWebClientId',
+            authRegion: 'testRegion',
+          },
+        },
+        [customOutputKey]: {
+          version: '1',
+          payload: {
+            customOutputs: JSON.stringify(customOutputs),
+          },
+        },
+      };
+      const outputRetrieval = mock.fn(async () => stubOutput);
+      const configContributors = [
+        new AuthClientConfigContributor(),
+        new CustomClientConfigContributor(),
+      ];
+
+      const clientConfigGenerator = new UnifiedClientConfigGenerator(
+        outputRetrieval,
+        configContributors
+      );
+      await assert.rejects(
+        () => clientConfigGenerator.generateClientConfig(),
+        (error: AmplifyUserError) => {
+          assert.strictEqual(
+            error.message,
+            'Duplicated entry with key aws_user_pools_id detected in deployment outputs'
+          );
```

Comment:
We've talked about this topic in design review session.
The conclusion was that:
1. We'll validate this for now strictly to see if there's demand for override scenario (since primary scenario is now that customer handcrafts a category using raw cdk, so maybe mixing our auth and their auth is not a good idea anyway).
2. If we get solid reasoning to add option to override, it would be something like `backend.addOutput({...}, { override: true})`. So that at least it's visible in their code that it was their intention and not a mistake.
3. If attempt to override is a mistake we'd rather fail fast than make customer debug it and look for warnings.